pkg/shell/base_index.js

```diff
@@ -481,6 +478,20 @@ function Router(index) {
 
             /* Add the child's group to all open channel messages */
             } else if (control.command == "open") {
+                // Prevent frames from malicious remote hosts from tampering with other machines
+                // The lookup may fail if a frame currently being removed
+                const child_host = source_by_name[child.name]?.default_host;
```

Comment:
Can the child change its own `.name`?

Reply:
This is an `<iframe>` property which the shell sets (i.e. the parent). If the iframe can change that (i.e. modify the DOM of its parent), then pretty much everything is lost anyway :grin: It is currently possible via running JS on its parent, as they both come from the same origin. This is yet another special case of "we need to sandbox the frames".